orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -52,6 +77,22 @@ Status Dropout<T1, T2>::ComputeInternal(OpKernelContext* context) const {
   //Get mask_data
   auto mask = context->Output(1, shape);
   ORT_ENFORCE(!mask || mask->Shape().Size() == N);
+
+  const Tensor* training_mode = context->Input<Tensor>(2);
+  //Check for inference mode.
+  if (!trainable_dropout && (training_mode == nullptr || *(training_mode->Data<bool>()) == false)) {
```

Comment:
TrainableDropout was using `ratio` input to decide between inference mode and training mode.  Specifically, `ratio == 0` indicated inference mode.  We seem to have lost that with these changes.  Can we keep that for backwards compatibility, until TrainableDropout is removed?

Reply:
Even if the RNG happens to do the right thing when ratio == 0 (I'm not sure) - this codepath is faster.

Reply:
Fair point.

---
In reply to: [419723580](https://github.com/microsoft/onnxruntime/pull/3805#discussion_r419723580) [](ancestors = 419723580)